plugins/modules/source_control/github/github_tag.py

```diff
@@ -0,0 +1,272 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright: Ansible Team
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: github_tag
+short_description: Create annotated GitHub tags for use with releases
+description:
+    - Allows for creation of annotated tags, by default a GitHub release only create lightweight tags.
+    - Use this module with the `github_release` module to created a release linked to a annotated tag.
+options:
+    url:
+        description:
+            - Custom GitHub URL for self-hosted enterprise versions.
+        type: str
+    token:
+        description:
+            - GitHub access Token for authenticating.
+            - Mutually exclusive with C(password).
+        type: str
+    username:
+        description:
+            - GitHub account username used for authentication.
+            - This is only needed when not using I(access_token).
+        type: str
+    password:
+        description:
+            - The GitHub account password for the user.
+            - This is only needed when not using I(access_token).
+        type: str
+    organization:
+        description:
+            - Repository organization name.
+        type: str
+        required: true
+    repo:
+        description:
+            - Repository name.
+        type: str
+        required: true
+    branch:
+        description:
+            - Name of the branch to create the tag on.
+        type: str
+        required: true
+    tag:
+        description:
+            - Name of the tag to create.
+        type: str
+        required: true
+    tagger:
+        description:
+            - Name of the tag to create.
+        type: dict
+        suboptions:
+            name:
+                description:
+                    - Name of the tagger
+                type: str
+                required: true
+            email:
+                description:
+                    - Email of the tagger
+                type: str
+                required: true
+        required: true
+    description:
+        description:
+            - description message for the tag to be created with.
+        type: str
+        required: true
+
+author:
+    - "Mark Woolley (@marknet15)"
+version_added: 3.9.0
+requirements:
+    - "github3.py >= 1.0.0"
+'''
+
+EXAMPLES = r'''
+- name: Create a annotated tag
+  community.general.github_tag:
+    token: tokenabc1234567890
+    organization: someorg
+    repo: testrepo
+    branch: main
+    tag: 1.0.0
+    tagger:
+      name: Some User
+      email: some.user@example.com
+    description: Some tag description message
+
+'''
+
+RETURN = r'''
+tag:
+    description:
+    - The tag name created
```

Comment:
```suggestion
    - The tag name created.
```
Same for the others.